src/XrdCl/XrdClAsyncSocketHandler.cc

```diff
@@ -220,11 +221,11 @@ namespace XrdCl
     {
       pLastActivity = time(0);
       if( unlikely( pTlsHandShakeOngoing ) )
-        OnTLSHandShake();
+        st = OnTLSHandShake();
```

Comment:
In reviewing this with Andy, we asked ourselves the question why on the parts where writing is concerned below, the status from `OnTLSHandShake()` and other calls is not being checked.

Reply:
For the call to OnTLSHandShake() below (concerning writing) the flow doesn't do anything after the call (i.e. it returns from AsyncSocketHandler::Event(), which is what I was going to do for the error case) so I didn't add the check. But it's a good point this looks quite strange perhaps it's a trap for future modifications, e.g. if something is added it might also get executed during the error condition.